docs/processes/release/tasks/healthy-code.md

```diff
@@ -477,7 +477,7 @@ steps:
     "source\\allinone\\allinone.sln".
 *   Build the Debug/Release + x86/x64 configurations (all 4 configurations).
     *   Make sure the generated data file (ex: "icu4c\\bin\\icudt64.dll") is not
-        stub data, it should be ~26MB.
+        stub data, it should be ~26MB. [Instructions on getting the full data](../../../userguide/icu4c/faq#why-am-i-seeing-a-small--only-a-few-k--instead-of-a-large--several-megabytes--data-shared-library-icudt)
```

Comment:
Does this need the `.md` extension (`faq.md`) for the link to work on GitHub?

Reply:
Good question, this was something I asked myself as well. And the answer for this and also the rest below is no :) it doesn't need the .md in the link.

The way I tested this is to go on the actual documentation website to whatever place you want to add or modify a link, inspect element on it and add the link exactly as you see it above, without the md extension. The link works and redirects you the place you want. 

This is also what the previous links that I'm replacing (and also the ones that work correctly) have, so I thought I would just keep it consistent

Reply:
Thanks Erik. Yeah, I guess the way the userguide site resolves the path differs from when navigating on GitHub.

For example, when viewing https://github.com/unicode-org/icu/blob/main/docs/userguide/icu4c/index.md on GitHub:
The "[ICU Architectural Design](https://github.com/unicode-org/icu/blob/main/docs/userguide/icu/design.md)" link directs us to the correct file since it has the extension, but the "[building ICU4C](https://github.com/unicode-org/icu/blob/main/docs/userguide/icu4c/build)" link is broken without the extension. I'm guessing this is because it's looking for the exact file name on GitHub.

So I'm not sure if it's worth fixing, but it seems like the docs are already a bit inconsistent with including the extension.

Reply:
Hmmmm interesting, well, if it makes it work in more places by adding the md extension, then I can totally do that :) 

thanks Daniel!

Reply:
The `.md` extension works well for people (ex: developers) looking at source files in the codebase.  But after Jekyll renders the Markdown files into the static HTML pages, it understands `.html` and no extension for URLs, but `.md` will cause a problem.  Since we care more about how the URLs for the Jekyll-rendered pages work, because they are end user-facing, I think we should consolidate on removing the `.md` if there's a choice.

I'll leave suggested edits in this PR where applicable.